### PR TITLE
Patch resources that cannot be replaced

### PR DESCRIPTION
**Pull request by kwiatekus**
**Related issues**
https://github.com/kyma-incubator/reconciler/issues/465

### CODE REVIEW

File: pkg/reconciler/kubernetes/internal/kubeclient.go [hunk 1]
```diff
@@ -132,24 +137,32 @@ func (kube *KubeClient) ApplyWithNamespaceOverride(u *unstructured.Unstructured,
...
 		return metadata, nil
 	}
 
-	replace := newReplace(helper)
-	replacedObject, err := replace(u, u.GetNamespace(), u.GetName())
-	if err != nil {
+	updateStrategyResolver := newDefaultUpdateStrategyResolver(helper)
+
+	strategy, err := updateStrategyResolver.Resolve(u)
+	if err != nil || strategy == SkipUpdateStrategy {
```
Comment (tobiscr) [inline pkg/reconciler/kubernetes/internal/kubeclient.go:146]: Why do we return an error when a SkipStragey was replied?

Comment (kwiatekus) [reply pkg/reconciler/kubernetes/internal/kubeclient.go:146]: I can separate those

Comment (kwiatekus) [reply pkg/reconciler/kubernetes/internal/kubeclient.go]: done

**Review by tobiscr — APPROVED** [review]

File: pkg/reconciler/kubernetes/internal/patcher.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+		lookupPatchMeta strategicpatch.LookupPatchMeta
+		schema          oapi.Schema
+	)
+	// createPatchErrFormat := "creating patch with:\noriginal:\n%s\nmodified:\n%s\ncurrent:\n%s\nfor:"
+
+	// Create the versioned types from the type defined in the restmapping
+	// (which is the API version we'll be submitting the patch to)
+	versionedObject, err := scheme.Scheme.New(p.Mapping.GroupVersionKind)
+
+	switch {
```
Comment (m00g3n) [inline pkg/reconciler/kubernetes/internal/patcher.go:132]: I know this part was reverted, however we could consider a refactor of the entire switch to be easier to read and maintain (we could have a separate story for this).

**Review by m00g3n — APPROVED** [review]
Got just 2 comments regarding the `patcher` type but we can address them in a follow-up task.

**Pull request merged**